File: app/src/main/java/com/sulistyo/moviecatalogueapi/adapter/java/MovieAdapter.java
package com.sulistyo.moviecatalogueapi.adapter.java;

import android.content.Context;
import android.content.Intent;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.LinearLayout;
import android.widget.TextView;
import androidx.recyclerview.widget.RecyclerView;
import com.bumptech.glide.Glide;
import com.sulistyo.moviecatalogueapi.R;
import com.sulistyo.moviecatalogueapi.model.movie.Movie;
import com.sulistyo.moviecatalogueapi.ui.activity.java.DetailMovieActivity;

import java.util.ArrayList;

public class MovieAdapter extends RecyclerView.Adapter<MovieAdapter.ViewHolder> {

    private Context context;
    private ArrayList<Movie> movies;

    private ArrayList<Movie> getMovies() {
        return movies;
    }

    public void setMovie(ArrayList<Movie> movies) {
        this.movies = movies;
    }

    public MovieAdapter(Context context) {
        this.context = context;
    }


    @Override
    public ViewHolder onCreateViewHolder(ViewGroup parent, int i) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.item, parent, false);
        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(ViewHolder viewHolder, final int position) {
        viewHolder.tvJudul.setText(getMovies().get(position).getTitle());
        Glide.with(context)
                .load(getMovies().get(position).getPhoto())
                .into(viewHolder.ivPoster);

        viewHolder.linearLayout.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Movie movi = new Movie();
                movi.setTitle(getMovies().get(position).getTitle());
                movi.setDesc(getMovies().get(position).getDesc());
                movi.setPhoto(getMovies().get(position).getPhoto());

                Intent i = new Intent(context, DetailMovieActivity.class);
                i.putExtra(DetailMovieActivity.EXTRA_MOVIE, movi);
                context.startActivity(i);
            }
        });
    }

    @Override
    public int getItemCount() {
        return getMovies().size();
    }

    class ViewHolder extends RecyclerView.ViewHolder {
        ImageView ivPoster;
        LinearLayout linearLayout;
        TextView tvJudul;

        ViewHolder(View itemView) {
            super(itemView);
            ivPoster = itemView.findViewById(R.id.ivPoster);
            linearLayout = itemView.findViewById(R.id.layout);
        }
    }
}

File: app/src/main/java/com/sulistyo/moviecatalogueapi/adapter/MovieAdapter.kt
package com.sulistyo.moviecatalogueapi.adapter

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.bumptech.glide.Glide
import com.sulistyo.moviecatalogueapi.R
import com.sulistyo.moviecatalogueapi.helper.networking.ApiCall
import com.sulistyo.moviecatalogueapi.model.movie.DataMovie
import kotlinx.android.synthetic.main.item_data.view.*

class MovieAdapter(private var mData: List<DataMovie>, val listener: (DataMovie) -> Unit) :
    RecyclerView.Adapter<MovieAdapter.vHolder>() {
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int) = vHolder(
        LayoutInflater.from(parent.context).inflate(R.layout.item_data, parent, false)
    )

    fun updateData(mData:List<DataMovie>){
        this.mData = mData
        notifyDataSetChanged()
    }

    override fun getItemCount() = mData.size

    override fun onBindViewHolder(holder: vHolder, position: Int) = holder.bind(mData.get(position), listener)


    class vHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        fun bind(data: DataMovie, listen: (DataMovie) -> Unit) {
            Glide.with(itemView.context).load(ApiCall.img + data.posterPath).into(itemView.ivPoster)
            itemView.tvJudul.text = data.title
            itemView.setOnClickListener { listen(data) }
        }
    }
}